src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -14,10 +18,18 @@
  */
 final class MaskingState implements ValueMaskerContext {
     private static final int INITIAL_JSONPATH_STACK_CAPACITY = 16; // an initial size of the jsonpath array
-    private final byte[] message;
+    private static final String STREAM_READ_ERROR_MESSAGE = "Failed to read input stream";
+    private static final String STREAM_WRITE_ERROR_MESSAGE = "Failed to write to output stream";
+
+    private byte[] message;
+    int messageLength;
+    private int bufferSize = 8192; // size of byte array buffers to be read from the input stream
```

Comment:
I was thinking, wouldn't it be easy to have this changeable (in a package private method) such that you can easily test any combination of overlap over multiple buffers? For example, set the buffer to 100 and have tons of hardcoded tests that have overlaps of buffers just before the colon, just after the colon, in the middle of a key, in the middle of an array, object, etc. etc.

Reply:
Good point. I think the best way is to make the field package-private and set it to small enough value in unit tests to cover scenarious below.

Reply:
Done